libbeat/common/kafka/sasl.go

```diff
@@ -54,23 +54,15 @@ func (c *SaslConfig) ConfigureSarama(config *sarama.Config) error {
 			return &XDGSCRAMClient{HashGeneratorFcn: SHA512}
 		}
 	default:
-		return fmt.Errorf("not valid mechanism '%v', only supported with PLAIN|SCRAM-SHA-512|SCRAM-SHA-256", c.SaslMechanism)
+		panic(fmt.Sprintf("not valid SASL mechanism '%v', only supported with PLAIN|SCRAM-SHA-512|SCRAM-SHA-256", c.SaslMechanism))
```

Comment:
Nit. Please add a comment here about why we are using panic. Something like this.
```suggestion
		// This should never happens because `SaslMechanism` is checked on `Validate()`, keeping a panic to detect it earlier if it happens.
		panic(fmt.Sprintf("not valid SASL mechanism '%v', only supported with PLAIN|SCRAM-SHA-512|SCRAM-SHA-256", c.SaslMechanism))
```